dask_match/tests/test_core.py

```diff
@@ -147,8 +152,20 @@ def test_conditionals(func):
     assert_eq(func(df), func(ddf))
 
 
-def test_predicate_pushdown():
-    df = ReadParquet("myfile.parquet", columns=("a", "b", "c"))
+@pytest.mark.xfail(reason="TODO: Debug this")
+def test_predicate_pushdown(tmpdir):
+    from dask_match.io.parquet import ReadParquet
+
+    fn = os.path.join(str(tmpdir), "myfile.parquet")
+    pd.DataFrame(
+        {
+            "a": [1, 2, 3, 4, 5] * 10,
+            "b": [0] * 50,
+            "c": range(50),
+        }
+    ).to_parquet(fn)
+
+    df = read_parquet(fn, columns=("a", "b", "c"))
```

Comment:
Can I ask for the following?

```suggestion
    original = pd.DataFrame(
        {
            "a": [1, 2, 3, 4, 5] * 10,
            "b": [0] * 50,
            "c": range(50),
        }
    )
    original.to_parquet(fn)

    df = read_parquet(fn)
    assert_eq(df, original)
```

My goal here is to verify that the graph generation works well

Reply:
Just a note that the graph logic works but this test is still marked as xfail, since I didn't get a chance to debug the predicate-pushdown behavior just yet.

Reply:
Ah, thanks for calling this out.  I see what's happening.  This class has a `.columns` attribute, and so that's taking precedence over `._meta.columns`.

The current system puts all of the parameters directly onto the object, which will inevitably conflict with other names, like column names or key attributes (as it does here).

The long-term solution is to have a wrapper class that handles user interaction and then hide all of the expression attributes inside an expr class.  This is also necessary for things like mutation.  This is good motivation for me to think about that class.  I think that you can safely leave this problem with me if you want.

Reply:
Or, for a short term fix, we would rename the `columns=` parameter to something else

Reply:
OK, things work on main now

Reply:
Cool - Thanks. It looks like we need a few more changes to make the computed result correct. About to submit something now.